src/QsCompiler/SyntaxProcessor/ScopeTools.fs

```diff
@@ -285,19 +285,23 @@ type SymbolTracker(globals : NamespaceManager, sourceFile, parent : QsQualifiedN
 
 /// The context used for symbol resolution and type checking within the scope of a callable.
 type ScopeContext =
-    { /// The symbol tracker for the parent callable.
+    { /// The namespace manager for global symbols.
+      Globals : NamespaceManager
+
+      /// The symbol tracker for the parent callable.
       Symbols : SymbolTracker
+
       /// True if the parent callable for the current scope is an operation.
       IsInOperation : bool
-      /// True if the current expression is contained within the condition of an if- or elif-statement.
-      IsInIfCondition : bool
```

Comment:
It would be good to formally deprecate things when reasonable. How much unused code do you need to keep to properly deprecate this? And along with it WithinIfCondition.

Reply:
Not much unused code. Re-added and marked obsolete.